packages/dx-react-grid/docs/guides/custom-plugin-development.md

```diff
@@ -0,0 +1,709 @@
+# React Grid - Custom Plugin Development
+
+
+## Table of Contents
+1. [Introduction](#introduction)
+2. [Render Custom Markup](#render-custom-markup)
+3. [Share Values Between Plugins](#share-values-between-plugins)
+4. [Use the Functionality of Built-In Plugins](#use-the-functionality-of-built-in-plugins)
+5. [Final Steps](#final-steps)
+
+
+## Introduction
+
+This tutorial is aimed at advanced users who are familiar with [basic Grid concepts](./getting-started.md) and who want to add new functionality to the Grid by using the [React Core](../../../core/docs/guides/fundamentals.md) components.
+
+In this tutorial, we add to the Grid's toolbar a select box that filters the grid by a predefined column. The purpose of the tutorial is not to implement a valuable feature but to show how to extend the Grid's functionality.
+
+We start with the following code. As you can see, we use the Grid in the `Material-UI` theme.
```

Comment:
Again, a very strange way to explain a simple idea. "We start with the following code. As you can see, we use the Grid in the `Material-UI` theme." -> In the following code, we use the Grid in the `Material-UI` theme: